cvmfs/sync_mediator.cc

```diff
@@ -226,6 +226,8 @@ manifest::Manifest *SyncMediator::Commit() {
     return NULL;
   }
 
+  if (catalog_manager_->IsBalanced())
```

Comment:
Should this be `if (!catalog_manager_->IsBalanced())`?  If so, how could this slip through the tests?